server/src/main/java/org/elasticsearch/common/settings/KeyStoreWrapper.java

```diff
@@ -310,7 +310,12 @@ public boolean hasPassword() {
     private Cipher createCipher(int opmode, char[] password, byte[] salt, byte[] iv) throws GeneralSecurityException {
         PBEKeySpec keySpec = new PBEKeySpec(password, salt, KDF_ITERS, CIPHER_KEY_BITS);
         SecretKeyFactory keyFactory = SecretKeyFactory.getInstance(KDF_ALGO);
-        SecretKey secretKey = keyFactory.generateSecret(keySpec);
+        SecretKey secretKey;
+        try {
+            secretKey = keyFactory.generateSecret(keySpec);
+        } catch (AssertionError ae) {
```

Comment:
I went back and forth as to how wide the net should be. My counter argument is that when something throws an Error instead of an Exception it supposedly denotes an irrecoverable error. We _know_ that this is not true for the AssertionError that BCFIPS throws here, but can we be certain that we want to catch all `Throwable`s that can be thrown from external code. We don't do that in any other place. WDYT ?

Reply:
You got me thinking again @ywangd .. On the one hand, I tried to limit the scope of these try-catch blocks to a min so it makes sense that we would catch all irrecoverable errors from _that_ line and give out an error message. On the other hand, the very limited scope makes it less possible that a wider net would catch issues thrown by other security providers that we know nothing of (other than the PBKDF2 112 bits limitation). I'm a bit torn on this one

Reply:
Had a short chat with Yang and we decided that catching `Error` is the way to go based on the arguments that have been made already in this PR